File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
# 11
# 1 4
# 3 5
# 0 6
# 5 7
# 3 8
# 5 9
# 6 10
# 8 11
# 8 12
# 2 13
# 12 14

n = int(input())
data = []

for i in range(n):
  s, e = map(int,input().split())
  data.append([s,e])

data.sort(key=lambda x:(x[1], x[0]))

count = 1

end = data[0][1]
for i in range(1, len(data)):
  if data[i][0] >= end :
    end = data[i][1]
    count = count + 1
  
print(count)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {
        //입력부
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int[][] data = new int[n][2];
        for(int i=0; i<n; i++) {
            st = new StringTokenizer(br.readLine());
            int start = Integer.parseInt(st.nextToken());
            int end = Integer.parseInt(st.nextToken());
            data[i][0] = start;
            data[i][1] = end;
        }

        //로직

        //빨리 끝나는 순으로 오름차순 정렬
        Arrays.sort(data, (o1, o2) -> {
            if(o1[1] == o2[1]) {
                return o1[0] - o2[0];
            }
            return o1[1] - o2[1];
        });

        int prevEndTime = 0;
        int count = 0;
        for(int[] time : data) {
            int nowStartTime = time[0];
            int nowEndTime = time[1];
            if(prevEndTime <= nowStartTime) {
                count++;
                prevEndTime = nowEndTime;
            }
        }

        System.out.println(count);
    }
}
